src/software/amazon/ion/impl/IonReaderTextSystemX.java

```diff
@@ -72,6 +73,46 @@ public IonSystem getSystem()
         return _system;
     }
 
+    // TODO getIntegerType() is duplicated in IonReaderBinarySystemX. It could
+    // be consolidated into a single location, but that would have to be part
```

Comment:
Moved back to an enum based on feedback, which allowed this to be consolidated in PrivateScalarConversions.